corehq/apps/userreports/views.py

```diff
@@ -87,98 +93,163 @@ def dispatch(self, request, domain, **kwargs):
+            'property_options': self.report_form.data_source_properties.values(),
+            'initial_filters': self.report_form.initial_filters,
+            'initial_columns': getattr(self.report_form, 'initial_columns', []),
         }
         return context
 
     @property
     @memoized
+    def configuration_form_class(self):
+        if self.existing_report:
+            type_ = self.existing_report.configured_charts[0]['type']
+        else:
+            type_ = self.request.GET.get('chart_type')
+        return {
+            'multibar': ConfigureBarChartReportForm,
+            'bar': ConfigureBarChartReportForm,
+            'pie': ConfigurePieChartReportForm,
+        }[type_]
+
+    @property
+    @memoized
     def report_form(self):
-        app_id = self.request.GET.get('application', '')
-        source_type = self.request.GET.get('source_type', '')
-        report_source = self.request.GET.get('source', '')
-        args = [app_id, source_type, report_source]
+        args = [self.request.GET.get(f, '') for f in self.url_args] + [self.existing_report]
         if self.request.method == 'POST':
             args.append(self.request.POST)
-        return self.configure_report_form_class(*args)
+        return self.configuration_form_class(*args)
 
     def post(self, *args, **kwargs):
         if self.report_form.is_valid():
-            report_configuration = self.report_form.create_report()
+            if self.report_form.existing_report:
+                report_configuration = self.report_form.update_report()
+            else:
+                report_configuration = self.report_form.create_report()
             return HttpResponseRedirect(
-                reverse(
-                    ConfigurableReport.slug,
-                    args=[self.domain, report_configuration._id]
-                )
+                reverse(ConfigurableReport.slug, args=[self.domain, report_configuration._id])
             )
         return self.get(*args, **kwargs)
 
 
-class ConfigurePieChartReportBuilderView(ConfigureBarChartReportBuilderView):
-    configure_report_form_class = ConfigureNewPieChartReport
-    report_title = _("Create New Report > Configure Pie Chart Report")
+class ConfigureListReport(ConfigureChartReport):
+    report_title = _("List Report: {}")
+
+    @property
+    @memoized
+    def configuration_form_class(self):
+        return ConfigureListReportForm
+
+
+class ConfigureTableReport(ConfigureChartReport):
+    report_title = _("Table Report: {}")
 
+    @property
+    @memoized
+    def configuration_form_class(self):
+        return ConfigureTableReportForm
 
-class ConfigureTableReportBuilderView(ConfigureBarChartReportBuilderView):
-    configure_report_form_class = ConfigureNewTableReport
-    report_title = _("Create New Report > Configure Table Report")
+
+class ConfigureWorkerReport(ConfigureChartReport):
+    report_title = _("Worker Report: {}")
```

Comment:
Hmm thinking about it more I guess it just treats the whole thing as a translation string, so as long as the translation also has `{}` in it, everything's fine